src/python/grpcio/grpc/__init__.py

```diff
@@ -348,6 +348,82 @@ def details(self):
         raise NotImplementedError()
 
 
+##############  Invocation-Side Interceptor Interfaces & Classes  ##############
+
+
+class UnaryClientInfo(six.with_metaclass(abc.ABCMeta)):
```

Comment:
I think it might be simpler to just invoke the callback with this data directly, rather than trying to wrap some of it up in a class.